### PR TITLE
Infrastructure: Finalize release automation script

### LINKED ISSUE #6424: Release Automation Tweaks
## Task Description

* [x] Improve pre-flight checks
  * [x] Check for changelog entry in `readme.txt`
* [x] Fix bash commands
* [x] (Post-)release version bumps on `main`
* [x] Figure out how to trigger Firebase deployment for static assets ([as per known limitation](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token))
* [x] SVN user for deployments
* [x] Documentation

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

There were a few remaining issues and questions around release automation.

## Summary

Addresses remaining issues after multiple rounds of testings.

Besides a few bugfixes and general workflow improvements now includes proper version bumps after release.

**How this has been tested**

* Setup a SVN repository at https://sourceforge.net/p/web-stories-wp/code/ to test deployment to WordPress.org
* Forked repository at https://github.com/swissspidy/web-stories-wp to test release workflow

Example releases: 

1.5.0 RC1: https://github.com/swissspidy/web-stories-wp/releases/tag/v1.5.0-rc.1
1.5.0 stable: https://github.com/swissspidy/web-stories-wp/releases/tag/v1.5.0

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6424

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -1,9 +1,7 @@
 === Web Stories ===
 
 Contributors:      google
-Requires at least: 5.3
 Tested up to:      5.7
-Requires PHP:      5.6
```
Comment (swissspidy) [inline readme.txt:6]: These two are no  longer needed here as per https://twitter.com/SergeyBiryukov/status/1370304657915060231

File: .github/workflows/plugin-release.yml [hunk 2]
```diff
@@ -315,23 +341,50 @@ jobs:
...
       - name: Update assets version
         run: npm run workflow:assets-version $ASSETS_VERSION
         if: ${{ env.ASSETS_VERSION }}
         env:
           ASSETS_VERSION: ${{ steps.assets_version.outputs.ASSETS_VERSION }}
 
-      - name: Commit changes
+      - name: Commit assets version bump
+        run: |
+          git add .
```
Comment (dreamofabear) [inline .github/workflows/plugin-release.yml]: Should we be more precise/less error prone? 

`git add web-stories.php`

File: .github/workflows/plugin-release.yml [hunk 3]
```diff
@@ -315,23 +341,50 @@ jobs:
...
+        if: ${{ env.ASSETS_VERSION }}
+        env:
+          ASSETS_VERSION: ${{ steps.assets_version.outputs.ASSETS_VERSION }}
+
+      - name: Update plugin version
+        run: npm run workflow:version $PLUGIN_VERSION
+
+      - name: Commit plugin version bump
         run: |
           git add .
```
Comment (dreamofabear) [inline .github/workflows/plugin-release.yml]: Ditto.

**Review by dreamofabear — APPROVED** [review]
Is there a "dry run" option that skips actual git commits/deploy steps but prints the commands instead? Would be nice for making changes to this script in the future.

**Comment by swissspidy** [comment]
> Is there a "dry run" option that skips actual git commits/deploy steps but prints the commands instead? Would be nice for making changes to this script in the future.

No, but good suggestion.

**Pull request merged**